lib/metadata_gen.js

```diff
@@ -10,11 +10,16 @@ class MetadataGenerator {
    * @param {PRData} data
    */
   constructor(data) {
-    const { owner, repo, pr, reviewers } = data;
+    const { owner, repo, pr, reviewers, commits } = data;
     this.owner = owner;
     this.repo = repo;
     this.pr = pr;
     this.reviewers = reviewers;
+    this.op = pr.bodyHTML;
+
+    commits.forEach(commit => {
+      this.op += commit.commit.messageBodyHTML;
```

Comment:
If there are multiple commits in one PR, a refs in commit A does not necessarily means it is referenced by commit B...need to figure out how to make this work

Reply:
So here i guess you mean some times a PR's commits needs to land in not just one but multiple commits (mostly 2-3 commits) so we want to output them differently, per commit.

Reply:
@cPhost Yes. Thinking about that, the best we could do is printing these differently for different commits, but don't write them into the actual output consumed by git-node

Reply:
> but don't write them into the actual output consumed by git-node

This a part i am bit confused on where exactly to print them (sure not it generated metadata) right.

Reply:
@cPhost we could do some refactoring with the link parser and get it to return those links separately, say one for the OP, and a map for each commit.

Reply:
@joyeecheung if you want to takeover this PR go ahead. I think you will be the right person to solve this. I cannot work on this anymore until after March 10 i got test to prepare for :)

Reply:
@cPhost Sure, I also plan to write some tests for git-node first so patches can be properly tested.

Good luck on your tests!

Reply:
Thanks.